### PR TITLE
refactor: Extract the status bar items to a centralized one

### PR DESCRIPTION
**Pull request by jdneo**
address #1506 

Signed-off-by: Sheng Chen <sheche@microsoft.com>

### CODE REVIEW

File: src/standardLanguageClient.ts [hunk 1]
```diff
@@ -35,24 +36,18 @@ export class StandardLanguageClient {
...
-
 		if (workspace.getConfiguration().get("java.showBuildStatusOnStart.enabled")) {
 			commands.executeCommand(Commands.SHOW_SERVER_TASK_STATUS);
 		}
 
 		serverStatus.initialize();
 		serverStatus.onServerStatusChanged(status => {
 			if (status === ServerStatusKind.Busy) {
-				item.text = '$(sync~spin)';
+				serverStatusBarProvider.updateText('$(sync~spin)');
```
Comment (fbricon) [inline src/standardLanguageClient.ts:46]: serverStatusBarProvider.setBusy()

File: src/standardLanguageClient.ts [hunk 2]
```diff
@@ -35,24 +36,18 @@ export class StandardLanguageClient {
...
 			} else if (status === ServerStatusKind.Error) {
-				item.text = '$(thumbsdown)';
+				serverStatusBarProvider.updateText('$(thumbsdown)');
```
Comment (fbricon) [inline src/standardLanguageClient.ts:48]: serverStatusBarProvider.setError()

File: src/standardLanguageClient.ts [hunk 3]
```diff
@@ -35,24 +36,18 @@ export class StandardLanguageClient {
...
 			} else {
-				item.text = '$(thumbsup)';
+				serverStatusBarProvider.updateText('$(thumbsup)');
```
Comment (fbricon) [inline src/standardLanguageClient.ts:50]: serverStatusBarProvider.setRunning()

Comment (jdneo) [reply src/standardLanguageClient.ts:50]: how about `setReady`?

Comment (fbricon) [reply src/standardLanguageClient.ts:50]: yes, even better

**Comment by fbricon** [comment]
Thanks @jdneo !

**Pull request merged**